### PR TITLE
chore: add git_hooks lib for optional use

### PR DESCRIPTION
**Pull request by jyeshe**
## What

Adds lib for optional git hooks install and checks.

## Why

Anticipate CI errors.

## Validation steps

0. Create a branch from this
1. `mix deps.get`
2. `mix git_hooks.install`
3. Try to make a commit with invalid elixir formatting
4. Fix the formatting
5. Commit a code with warning and try to push it
6. Fix the warning
7. Commit a code that doesn't pass a unit test and try to push it
8. Fix the unit test or the code
9. Commit and push (are successful)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4802,3 +4805,24 @@ The example output would look like:
...
+
+#### Actions
+
+On push:
+- Commit linter for conventional commit messages
+- Elixir code formatting
+- Dialyzer
+- Unit tests
+
+On merge to master:
```
Comment (jyeshe) [inline README.md:4819]: this shall be on tag push

**Review by sborrazas — APPROVED** [review]
We should probably leave automated local dev tools to be set individually, but since this is optional :+1:

**Comment by jyeshe** [comment]
> We should probably leave automated local dev tools to be set individually, but since this is optional +1

dev.tools.exs we have (:

**Review by sborrazas — APPROVED** [review]
Thanks!

**Review by thepiwo — APPROVED** [review]

**Pull request merged**